rust/otap-dataflow/crates/otlp/src/fake_signal_receiver/fake_signal.rs

```diff
@@ -0,0 +1,466 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//!
+//! The fake signal module provides methods for generating OTLP signals for testing
+//!
+
+use crate::fake_signal_receiver::config::MetricType;
+use crate::fake_signal_receiver::fake_data::*;
+use crate::proto::opentelemetry::{
+    common::v1::InstrumentationScope,
+    logs::v1::{LogRecord, LogsData, ResourceLogs, ScopeLogs},
+    metrics::v1::{
+        Exemplar, ExponentialHistogram, ExponentialHistogramDataPoint, Gauge, Histogram,
+        HistogramDataPoint, Metric, MetricsData, NumberDataPoint, ResourceMetrics, ScopeMetrics,
+        Sum, Summary, SummaryDataPoint, exponential_histogram_data_point::Buckets, metric::Data,
+        summary_data_point::ValueAtQuantile,
+    },
+    profiles::v1development::{Profile, ProfilesData, ResourceProfiles, ScopeProfiles},
+    resource::v1::Resource,
+    trace::v1::{
+        ResourceSpans, ScopeSpans, Span, TracesData,
+        span::{Event, Link},
+    },
+};
+
+/// Genererates MetricsData based on the provided count for resource, scope, metric, datapoint count
+#[must_use]
+pub fn fake_otlp_metrics(
+    resource_metrics_count: usize,
+    scope_metrics_count: usize,
+    metric_count: usize,
+    datapoint_count: usize,
+    datapoint_type: MetricType,
+    attribute_count: usize,
+) -> MetricsData {
+    let mut resource_metrics: Vec<ResourceMetrics> = vec![];
+
+    for _ in 0..resource_metrics_count {
+        let mut scope_metrics: Vec<ScopeMetrics> = vec![];
+        for _ in 0..scope_metrics_count {
+            // create some fake metrics based on the metric_count value provided
+            let mut metrics: Vec<Metric> = vec![];
+            for _ in 0..metric_count {
+                metrics.push(fake_metric(
+                    datapoint_type,
+                    datapoint_count,
+                    attribute_count,
+                ));
+            }
+            scope_metrics.push(ScopeMetrics {
+                schema_url: get_schema_url(),
+                scope: Some(InstrumentationScope {
+                    name: get_scope_name(),
+                    version: get_scope_version(),
+                    attributes: get_attributes(attribute_count),
+                    dropped_attributes_count: 0,
+                }),
+                metrics: metrics.clone(),
+            });
+        }
```

Comment:
I'm still wondering if we want to continue with the builder macros developed in `otel-arrow-rust/src/pdata/otlp`. If we did, this would be written more like:
```
scope_metrics.push(ScopeMetrics::build(
    InstrumentationScope::build(get_scope_name())
    .version(get_scope_version())
        .attributes(get_scope_attributes())
        .finish())
     .metrics(metrics)
     .finish());
```

I feel "on the fence" about this approach. Now that you've written many Prost OTLP structs out by hand, how do you feel?